src/pages/Shop/Shop.jsx

```diff
@@ -1,21 +1,122 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import styled from 'styled-components';
 
 import Category from './Category/Category';
 import ShopHeader from './ShopHeader/ShopHeader';
 import SideFilterList from './SideFilterList/SideFilterList';
 import ProductList from './ProductList/ProductList';
+import { PRODUCTS_API } from '../../config';
+import { PRODUCTS_CATEGORY_API } from '../../config';
 
 const Shop = () => {
+  const [productList, setProductList] = useState([]);
+  const [filterList, setFilterList] = useState([]);
+  const [queryState, setQueryState] = useState([]);
+  const [sortList, setSortList] = useState([]);
+
+  const sortDrop = e => {
+    setSortList(e.target.value);
+  };
+
+  console.log(`sortList`, `${sortList}`);
+
+  useEffect(() => {
+    fetch(`${PRODUCTS_API}?${addQuery}&${sortList}`)
+      // fetch(`${PRODUCTS_API}?author=1`)
+      .then(res => res.json())
+      .then(data => {
+        setProductList(data.results);
+      });
+  }, [queryState, sortList]);
```

Comment:
- addQuery 는 여기에 위치하는게 더 적절할 것 같습니다.
```suggestion
  useEffect(() => {
    const addQuery = queryState.map(state => makeQuery(state)).join('');
    fetch(`${PRODUCTS_API}?${addQuery}&${sortList}`)
      .then(res => res.json())
      .then(data => {
        setProductList(data.results);
      });
  }, [queryState, sortList]);
```

Reply:
수정했습니다.